### PR TITLE
backupinfo: introduce a backupinfo package

### PR DESCRIPTION
**Pull request by adityamaru**
The backupinfo package contains logic related to interacting
with information and metadata describing the backup. After this
change we have `backupdest` depending on `backupinfo`.

Release note: None

### CODE REVIEW

File: pkg/ccl/backupccl/backup_job.go [hunk 1]
```diff
@@ -296,8 +297,8 @@ func backup(
 			// Set a unique filename for each partition backup descriptor. The ID
 			// ensures uniqueness, and the kv string appended to the end is for
 			// readability.
-			filename := fmt.Sprintf("%s_%d_%s",
-				backupPartitionDescriptorPrefix, nextPartitionedDescFilenameID, sanitizeLocalityKV(kv))
+			filename := fmt.Sprintf("%s_%d_%s", backupPartitionDescriptorPrefix,
+				nextPartitionedDescFilenameID, backupinfo.SanitizeLocalityKV(kv))
```
Comment (stevendanna) [inline pkg/ccl/backupccl/backup_job.go:301]: [future] We use "Sanitize" in some places to mean "redaction" and then here we use it to mean "make a name suitable to be a filename".  We can probably clean up the naming here.

File: pkg/ccl/backupccl/backupinfo/manifest_handling.go [hunk 2]
```diff
@@ -51,40 +54,38 @@ import (
 
 // Files that may appear in a backup directory.
 const (
-	// backupManifestChecksumSuffix indicates where the checksum for the manifest
+	// BackupManifestChecksumSuffix indicates where the checksum for the manifest
 	// is stored if present. It can be found in the name of the backup manifest +
 	// this suffix.
-	backupManifestChecksumSuffix = "-CHECKSUM"
+	BackupManifestChecksumSuffix = "-CHECKSUM"
```
Comment (stevendanna) [inline pkg/ccl/backupccl/backupinfo/manifest_handling.go:60]: [future,discussion] Some of these being public is a signal that perhaps we need some other methods in this package. That is, eventually do we expect it to be the case that most users of this package don't need to worry about the internal structure of these constants.

**Review by stevendanna — COMMENTED** [review]
Did an early read through since I'm thinking of also doing some reorganisation.

Overall, I think it makes sense to have a package focused on interacting with the context of a specific backup directory.

It's easy to spend a lot of time fretting about details (is a backup directory and the chain of incrementals related to it its own concept that deserve a name, should we separate the reading of the raw bytes out of the backup from parsing their meaning a bit more, etc).  But, this seems like a reasonable first cut to me.

**Comment by adityamaru** [comment]
@stevendanna I've made you wait a little but this should be RFAL 😃 I agree that there is definitely follow up cleanup that we should undertake including moving some of the constants in `backupbase` into relevant packages.

**Comment by livlobo** [comment]
@adityamaru  - mind filing a GH issue corresponding to this PR?

**Comment by adityamaru** [comment]
@livlobo filed https://github.com/cockroachdb/cockroach/issues/83338

**Comment by livlobo** [comment]
Perfect, thank you!

**Review by stevendanna — APPROVED** [review]
Nice.

**Pull request merged**